inductiva/templating/engine.py

```diff
@@ -0,0 +1,142 @@
+            render_args: Arguments to render the template file.
+        """
+
+        if not self.renderer.is_template(source_file):
+            raise ValueError(f"{source_file} is not a template file.")
+        source_file = self.__template_dir / source_file
+
+        if target_file is None:
+            target_file = pathlib.Path(source_file).name
+
+        target_file = self.renderer.strip_extension(target_file)
+        target_file = self.__root_dir / target_file
+
+        if not overwrite and target_file.exists():
+            raise FileExistsError(f"{target_file} already exists.")
+
+        (target_file.parent).mkdir(parents=True, exist_ok=True)
+        self.renderer.render_file(target_file=target_file,
+                                  source_file=source_file,
+                                  **render_args)
+
+        return target_file
+
+    def render_dir(self,
+                   source_dir: Optional[types.Path] = None,
+                   target_dir: Optional[types.Path] = None,
+                   overwrite: bool = False,
+                   **render_args):
+        """Render a directory to the root_dir and render all template files.
+        
+        Render the directory `source_dir` in the template directory to the
+        `target_dir` inside the root directory from the given `render_args`.
+        It keeps the structure from the source directory. 
+        
+        If `source_dir` is None, the root template_dir is used. If `target_dir`
+        is None, the source_dir is rendered into the root directory. All
+        template files will have the template suffix stripped from their name
+        upon copying.
+
+        In case, the names of the files being rendered to the target_dir already
+        exist, method by default raises a FileExistsError. To overwrite them,
+        set the flag `overwrite` to True.
+
+        Args:
+            source_dir (types.Path): Path to the source directory.
+            target_dir (types.Path): Path to the target directory.
+            overwrite (bool): Overwrite the target directory if  already exists.
+            render_args: Arguments to render the template files.
+        """
+
+        if source_dir is None:
+            source_dir = self.__template_dir
+
+        if target_dir is None:
+            target_dir = "."
+
+        target_dir = self.__root_dir / target_dir
+
+        if not overwrite:
+            try:
+                self._validate_destination(source_dir, target_dir)
+            except FileExistsError as e:
+                msg = f"{e}; set `overwrite=True` to overwrite existing files."
+                raise FileExistsError(msg) from e
+
+        self.renderer.render_dir(source_dir=source_dir,
+                                 target_dir=target_dir,
+                                 **render_args)
+
+        return target_dir
+
+    def _validate_destination(self, source_dir: pathlib.Path,
```

Comment:
There is an equivalent one in the FileManager, here we are specifying the ones with the Template Extension, since we want to check if there aren't any inside the target_dir already with the same as a template file without the extension. So, the reason to fail is slight different.